src/analyze/analyze-verify.h

```diff
@@ -6,5 +6,17 @@
 #include "execute.h"
 #include "path-lookup.h"
 
+typedef enum ReturnErrorOn {
+        RETURN_ERROR_ON_NONE,                                 /* Never returns errors on syntax warnings */
+        RETURN_ERROR_ON_ANY_WARNING,                          /* Returns errors when warnings arise in any associated unit */
+        RETURN_ERROR_ON_WARNING_IN_SELECTED,                  /* Returns errors when warnings arise in the selected unit */
+        RETURN_ERROR_ON_WARNING_IN_SELECTED_AND_DEPENDENCIES, /* Returns errors when warnings arise in the selected unit or its direct dependencies */
```

Comment:
this looks irritatingly complex? can't we find a single mode that works for everyone? i dislike options that are hard to understand for little gain. Do you really expect people to knowingly pick one of these 4 different values?

Reply:
The first one is to maintain backward compatibility - we can remove that if there's no intention to provide that option (not sure if anybody cares).

The second one is the new default, and it's for full-os images, and behaves as one expects - any error anywhere recursively causes a failure.

The third one is the one we absolutely need for our use case - an extension image, which means it has a small subset of units.

The last one was requested by @keszybz and he needs it for his use case.

So, at most the backward-compat option can be removed, but the other are fundamentally in conflict with each other I'm afraid

Reply:
i'd drop the backward compat. i'd consider the old behaviour a bug.

maybe rename this --recursive-errors= or so which takes a boolean or the special string "one" or so. because it's really just about the depth of recursion for looking for errros, no?

Reply:
So instead `--fail-verification-on`, have `--recursive-errors=yes|no|one`? @keszybz is that ok for you?

Reply:
OK.